nats/src/main.rs

```diff
@@ -229,11 +286,26 @@ impl ProviderHandler for NatsMessagingProvider {
     #[instrument(level = "info", skip(self))]
     async fn delete_link(&self, actor_id: &str) {
         let mut aw = self.actors.write().await;
-        if aw.remove(actor_id).is_some() {
-            info!("nats closing connection for actor {}", actor_id);
-            // close and drop the connection
-            // dropping the client should close it
-        } // else ignore: it's already been dropped
+
+        if let Some(NatsClientBundle { sub_handles, .. }) = aw.get(actor_id) {
```

Comment:
Change this to `aw.remove` up here. It will return `Option<T>` anyway and then everything will drop once you've used it. 

Actually, now that I think about it, you can do something like:

```rust
let bundle = {
    let mut aw = self.actors.write().await;
    aw.remove(actor_id)
};
```

Then you can do `if let Some()` here (like you are currently doing) without holding a lock while you clean up

Reply:
See [this comment](https://github.com/wasmCloud/capability-providers/pull/205/files#r1116488085) -- I explicitly made sure to go through the subscriptions lexically because I wanted to make absolutely sure that `.abort()` was called. 

[`drop()`ing a join handle detaches it](https://docs.rs/tokio/latest/src/tokio/runtime/task/join.rs.html#18) as far I can tell, it does not necessarily stop it.

Reply:
I think I miscommunicated. Your for loop will remain unchanged. But calling read to just get them _and then_ dropping them is superfluous. Remove is also a get, but also pulls it out of the hashmap in one go so you only need to lock once. Then you run through the same `abort` stuff after. If you want complete guarantees, you could wrap the `Vec` in another struct and impl `Drop` for that struct

Reply:
Ahhhh gotcha! thanks will update this